File: CodingSamples/DotNet/Fundamentals/Basics/InheritanceTest/InterfaceTypeTest/Banking/CurrentAccount.cs
namespace Banking;

//a class declared with sealed keyword cannot be used
//as a base class for other class
sealed class CurrentAccount : Account
{
    public override void Deposit(decimal amount)
    {
        Balance += amount;
    }

    public override void Withdraw(decimal amount)
    {
        if(Id < 0)
            throw new NotSupportedException("Account is locked.");
        Balance -= amount;
    }
}

File: CodingSamples/Java/Fundamentals/Inheritance/InterfaceTypeTest/bank/banking/CurrentAccount.java
package banking;

//a class defined with final keyword cannot be extended
final class CurrentAccount extends Account {
   
    public void deposit(double amount) {
        if(balance < 0)
            amount -= 500;
        balance += amount;
    }

    public void withdraw(double amount) throws InsufficientFundsException {
        balance -= amount;
    }
}

